File: Data_Structures/4_Queue/Python/queue.py
class Queue():

    # Construtor da classe
    def __init__(self):
        self.items = []

    # Função para verificar se a fila está vazia
    def estaVazia(self):
        return self.items == []
    
    # Função para adicionar um novo elemento na fila
    def enfileira(self, item):
        self.items.insert(0, item)

    # Função para remover um elemento da fila
    def desenfileira(self):
        return self.items.pop()
    
    # Função para retornar o tamanho da fila
    def tamanhoFila(self):
        return len(self.items)
    
    # Função para retornar o primeiro elemento da fila
    def frente(self):
        return self.items[len(self.items)-1]

File: Data_Structures/4_Queue/Rust/queue.rs
pub struct Queue {
    items: Vec<i32>,
}


impl Queue {
    pub fn new() -> Self {
        Queue { items: Vec::new() }
    }

    pub fn enfileira(&mut self, data: i32) {
        self.items.push(data);
    }

    pub fn desenfileira(&mut self) -> Option<i32> {
        if self.items.is_empty() {
            return None;
        }
        Some(self.items.remove(0))
    }

    pub fn frente(&self) -> Option<&i32> {
        self.items.first()
    }

    pub fn tamanho_fila(&self) -> i32 {
        self.items.len() as i32
    }

    pub fn esta_vazia(&self) -> bool {
        self.items.is_empty()
    }
}